# Pre Test Use Case
<br> **Telkomsel Data Science Academy**
<br> 16 October 2019
<br> Dataset :[https://www.openml.org/d/40701](https://www.openml.org/d/40701)
* * * 

### import library

In [24]:
import pandas as pd
import numpy as np
import statistics as st

### Load data into df

In [25]:
df=pd.read_csv('https://www.openml.org/data/get_csv/4965302/churn.arff')
df.head()

state  account_length  area_code  phone_number  international_plan  \
0     16             128        415          2845                   0   
1     35             107        415          2301                   0   
2     31             137        415          1616                   0   
3     35              84        408          2510                   1   
4     36              75        415           155                   1   

   voice_mail_plan  number_vmail_messages  total_day_minutes  total_day_calls  \
0                1                     25              265.1              110   
1                1                     26              161.6              123   
2                0                      0              243.4              114   
3                0                      0              299.4               71   
4                0                      0              166.7              113   

   total_day_charge  ...  total_eve_calls  total_eve_charge  \
0             45.07  ...               99             16.78   
1             27.47  ...              103             16.62   
2             41.38  ...              110             10.30   
3             50.90  ...               88              5.26   
4             28.34  ...              122             12.61   

   total_night_minutes  total_night_calls  total_night_charge  \
0                244.7                 91               11.01   
1                254.4                103               11.45   
2                162.6                104                7.32   
3                196.9                 89                8.86   
4                186.9                121                8.41   

   total_intl_minutes  total_intl_calls  total_intl_charge  \
0                10.0                 3               2.70   
1                13.7                 3               3.70   
2                12.2                 5               3.29   
3                 6.6                 7               1.78   
4                10.1                 3               2.73   

   number_customer_service_calls  class  
0                              1      0  
1                              1      0  
2                              0      0  
3                              2      0  
4                              3      0  

[5 rows x 21 columns]

### get info on data

In [29]:
df.info()
print('\nData Size \nfeatures : ',len(df.columns),'\nbaris : ',len(df))

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 21 columns):
state                            5000 non-null int64
account_length                   5000 non-null int64
area_code                        5000 non-null int64
phone_number                     5000 non-null int64
international_plan               5000 non-null int64
voice_mail_plan                  5000 non-null int64
number_vmail_messages            5000 non-null int64
total_day_minutes                5000 non-null float64
total_day_calls                  5000 non-null int64
total_day_charge                 5000 non-null float64
total_eve_minutes                5000 non-null float64
total_eve_calls                  5000 non-null int64
total_eve_charge                 5000 non-null float64
total_night_minutes              5000 non-null float64
total_night_calls                5000 non-null int64
total_night_charge               5000 non-null float64
total_intl_minutes               

#### Get Count Unique Value for each columns

In [30]:
df.nunique()

state                              51
account_length                    218
area_code                           3
phone_number                     5000
international_plan                  2
voice_mail_plan                     2
number_vmail_messages              48
total_day_minutes                1961
total_day_calls                   123
total_day_charge                 1961
total_eve_minutes                1879
total_eve_calls                   126
total_eve_charge                 1659
total_night_minutes              1853
total_night_calls                 131
total_night_charge               1028
total_intl_minutes                170
total_intl_calls                   21
total_intl_charge                 170
number_customer_service_calls      10
class                               2
dtype: int64

#### Feature
***
* phone_number is unique data accross the dataset
* state, area_code are nominal datatype
* class, international_plan, voice_mail_plan are binary datatype
* account_length, number_of_vmail_messages, number_customer_service_calls, and every total_* columns all probably measurement of data

#### Get info on class

In [22]:
df['class'].value_counts()

0    4293
1     707
Name: class, dtype: int64

* there is 4293 class 0 data
* there is 707 class 1 data

### What Central of  measurement to class

In [43]:
df_class = df.iloc[:,4:]
df_class.groupby('class',as_index=True).mean().transpose()

class                                   0           1
international_plan               0.063825    0.281471
voice_mail_plan                  0.284416    0.144272
number_vmail_messages            8.291870    4.496464
total_day_minutes              175.746564  207.870580
total_day_calls                 99.899604  100.817539
total_day_charge                29.877494   35.338416
total_eve_minutes              198.805031  211.757850
total_eve_calls                100.241556   99.884017
total_eve_charge                16.898654   17.999562
total_night_minutes            199.455113  206.078218
total_night_calls               99.975775   99.575672
total_night_charge               8.975593    9.273607
total_intl_minutes              10.190869   10.692362
total_intl_calls                 4.481947    4.151344
total_intl_charge                2.752055    2.887426
number_customer_service_calls    1.457722    2.254597

* Class 0 memiliki mean values International_plan, voice_mail_plan, number_vmail_messages yang lebih tinggi dibandingkan class 1
* sedangkan Class 0 memiliki nmber_customer_service_calls, total_day_minutes, total_day_charge yang lebih rendah dibandingkan dengan class 1
* untuk measurement lain relative memiliki mean yang sama

### What Measure of Dispersion from group class

In [54]:
df_class = df.iloc[:,4:]
df_class.groupby('class',as_index=True).std().transpose()

class                                  0          1
international_plan              0.244469   0.450035
voice_mail_plan                 0.451189   0.351614
number_vmail_messages          13.809408  11.297719
total_day_minutes              49.634575  68.576475
total_day_calls                19.674019  20.757110
total_day_charge                8.437810  11.658195
total_eve_minutes              50.151854  51.568859
total_eve_calls                19.862391  19.618258
total_eve_charge                4.262900   4.383293
total_night_minutes            50.609570  49.685763
total_night_calls              19.958820  19.968555
total_night_charge              2.277438   2.235902
total_intl_minutes              2.749994   2.793187
total_intl_calls                2.438233   2.550011
total_intl_charge               0.742443   0.754057
number_customer_service_calls   1.164236   1.815956

* Hampir semua measure memiliki dispersion yang relative mirip untuk class 0 dan 1
* hanya internation_plan, voice_mail_plan, total_day_minutes, dan number_customer_service_calls yang memiliki dispersion yang membedakan antara class 0 dan 1
* dengan kolom total_eve_minutes, total_day_minutes, total_night_minutes memiliki dispersion terbesar

### Correlation all measures to Class

In [52]:
df_class = df.iloc[:,4:]
abs(df_class.corr(method='kendall').loc['class'].sort_values(ascending=False))

class                            1.000000
international_plan               0.259123
total_day_charge                 0.138156
total_day_minutes                0.138156
number_customer_service_calls    0.130292
total_eve_charge                 0.069131
total_eve_minutes                0.069129
total_intl_charge                0.047158
total_intl_minutes               0.047158
total_night_charge               0.036332
total_night_minutes              0.036311
total_day_calls                  0.015960
total_night_calls                0.006459
total_eve_calls                  0.007171
total_intl_calls                 0.056519
number_vmail_messages            0.097714
voice_mail_plan                  0.110698
Name: class, dtype: float64

* korelasi tertinggi adalah international_plan dengan value 0.259, dan semua measurement features memiliki korelasi yang rendah